executor/common_linux.h

```diff
@@ -1095,7 +1095,15 @@ static long syz_mount_image(volatile long fsarg, volatile long dir, volatile uns
 {
 	char loopname[64], fs[32], opts[256];
 	int loopfd, err = 0, res = -1;
+	unsigned int loopdevno;
 	unsigned long i;
+	char* dir2 = NULL;
+	char dirtemplate[] = "/tmp/template.XXXXXX";
+
+	// TODO: Remove "dir" argument from syz_mount_image and use temporary directory created
+	// internally instead.
+	dir = 0;
```

Comment:
Is it fine to remove "dir" and create the temporary directory internally?

Reply:
No, it's not OK for the following reasons:
1. Tests must be self-contained. mkdtemp creates the directory in an unknown location. This will cause problems during reproduction.
2. The fuzzer does not know about the resulting directory, so it will not use it. So while it makes mounting of an image simpler, it does not allow to test any operations in the mounted image. And in this sense makes coverage of fs code worse.
3. /tmp may not exist, be writable, etc.

Fuzzer needs to give a local directory that it knows about. And  you need to create this dir in the test program too.

Reply:
I see, yes. In my first few attempts, I tried writing a test that created a directory, and call syz_mount_image
on that but ended up with an EBUSY. Initially I thought this was because of repeated mounts on the same
mountpoint(though it was not, discussed more below). I'll change this back to use the original dir, and look
to fixing any problems that arise.

Reply:
The more realistic the test, the better.
If mkdir+syz_mount_image does not work, then it also won't work during fuzzing. Which means we will not actually test it.